File: Sistema de Gestao Academico/Web.Server/Models/Turma.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Web.Server.Models
{
    public class Turma
    {
        public int Id { get; set; }
        [Required]
        public string Codigo { get; set; }
        [Required]
        public Periodos Periodo { get; set; }
        [Required]
        public int CursoId { get; set; }
        [Required]
        public int DisciplinaId { get; set; }
        [Required]
        public int SemestreId { get; set; }
        [Required]
        public int ProfessorId { get; set; }


        [NotMapped]
        public Curso Curso { get; set; }
        [NotMapped]
        public Disciplina Disciplina { get; set; }
        [NotMapped]
        public Semestre Semestre { get; set; }
        [NotMapped]
        public Usuario Professor { get; set; }
    }
}

File: Sistema de Gestao Academico/Web/ClientApp/src/app/_models/turma.ts
import { Periodos } from "../_enums/periodos.enum";
import { Curso } from "./curso";
import { Disciplina } from "./disciplina";
import { Semestre } from "./semestre";
import { Usuario } from "./usuario";

export class Turma {
  id: number;
  codigo: string;
  periodo: Periodos;
  cursoId: number;
  disciplinaId: number;
  semestreId: number;
  professorId: number;

  curso: Curso;
  disciplina: Disciplina;
  semestre: Semestre;
  professor: Usuario;
}
